lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -64,8 +64,18 @@ def buffers
 
     # NB https://github.com/zdennis/activerecord-import/wiki/On-Duplicate-Key-Update
     def self.persist_buffers!
-      column_keys = PolicyElement.column_names.map(&:to_sym)
-      PolicyElement.import(buffers[:upsert].values, on_duplicate_key_update: column_keys - [:id])
+      column_keys = PolicyElement.column_names
+
+      # Because activerecord-import cannot yet handle arbitrary serialized values
+      # during import, we set all attributes again in bulk here.  It is important
+      # that these changes are mutative, since the default ActiveRecord magic
+      # being relied on for assignments and associations will break, otherwise
+      elements_to_upsert = buffers[:upsert].values.map do |element|
```

Comment:
`buffers[:upsert].values.each { |element| element.attributes = element.attributes.slice(*column_keys)` seems clearer to me, makes it more obvious that it mutates. You could also definite an instance method on PolicyElement for this and make it
`values.each(&:prep_for_import!)`